applications/ParticleMechanicsApplication/custom_conditions/particle_based_conditions/mpm_particle_lagrange_dirichlet_condition.cpp

```diff
@@ -0,0 +1,458 @@
+#include "includes/checks.h"
+
+
+namespace Kratos
+{
+//******************************* CONSTRUCTOR ****************************************
+//************************************************************************************
+
+MPMParticleLagrangeDirichletCondition::MPMParticleLagrangeDirichletCondition( IndexType NewId, GeometryType::Pointer pGeometry )
+    : MPMParticleBaseDirichletCondition( NewId, pGeometry )
+{
+    //DO NOT ADD DOFS HERE!!!
+}
+
+//************************************************************************************
+//************************************************************************************
+
+MPMParticleLagrangeDirichletCondition::MPMParticleLagrangeDirichletCondition( IndexType NewId, GeometryType::Pointer pGeometry,  PropertiesType::Pointer pProperties )
+    : MPMParticleBaseDirichletCondition( NewId, pGeometry, pProperties )
+{
+}
+
+//********************************* CREATE *******************************************
+//************************************************************************************
+
+Condition::Pointer MPMParticleLagrangeDirichletCondition::Create(IndexType NewId,GeometryType::Pointer pGeometry,PropertiesType::Pointer pProperties) const
+{
+    return Kratos::make_intrusive<MPMParticleLagrangeDirichletCondition>(NewId, pGeometry, pProperties);
+}
+
+//************************************************************************************
+//************************************************************************************
+
+Condition::Pointer MPMParticleLagrangeDirichletCondition::Create( IndexType NewId, NodesArrayType const& ThisNodes,  PropertiesType::Pointer pProperties ) const
+{
+    return Kratos::make_intrusive<MPMParticleLagrangeDirichletCondition>( NewId, GetGeometry().Create( ThisNodes ), pProperties );
+}
+
+//******************************* DESTRUCTOR *****************************************
+//************************************************************************************
+
+MPMParticleLagrangeDirichletCondition::~MPMParticleLagrangeDirichletCondition()
+{
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void MPMParticleLagrangeDirichletCondition::InitializeSolutionStep( ProcessInfo& rCurrentProcessInfo )
+{
+    MPMParticleBaseDirichletCondition::InitializeSolutionStep( rCurrentProcessInfo );
+
+    GeometryType& r_geometry = GetGeometry();
+    const unsigned int number_of_nodes = r_geometry.PointsNumber();
+    const unsigned int dimension = r_geometry.WorkingSpaceDimension();
+
+    for ( unsigned int i = 0; i < number_of_nodes; i++ )
+    {
+        array_1d<double, 3 > & lagrange_multiplier  = r_geometry[i].FastGetSolutionStepValue(VECTOR_LAGRANGE_MULTIPLIER);
+
+        for ( unsigned int j = 0; j < dimension; j++ )
+        {
+            #pragma omp atomic
+            lagrange_multiplier[j] *= 0.0;
```

Comment:
I don't understand why you reset the LM at the beginning. If a condition continues setting a certain configuration the previous LM may be a good estimate. In my experience reset the LM is a bad practice, specially for contact problems

Besides, as a nodal variable the reset should not be done here, in the element

Reply:
The idea was, that if the condition particle moves to another background element, it is not correct, if the old background element keeps the value. 
But I will think about it again. Thanks

Reply:
I assume the movement is not instantaneous, and in any case probably if there is a particle there it will start with that LM which will be a good initial estimation. If the LM must be zero should not be enforced, the weak form will impose it by itself, except because you know that must be zero and therefore deactivate will improve convergence. For example, in a contact problem with active sets, and deactivating one of the nodes